source/lex.tex

```diff
@@ -1033,28 +1035,38 @@
 \indextext{literal!\idxcode{unsigned}}%
 \indextext{literal!\idxcode{long}}%
 \indextext{literal!base of integer}%
-An \defnx{integer literal}{literal!integer} is a sequence of digits that has no period
-or exponent part, with optional separating single quotes that are ignored
-when determining its value. An integer literal may have a prefix that specifies
-its base and a suffix that specifies its type. The lexically first digit
-of the sequence of digits is the most significant.
-A \defnx{binary integer literal}{literal!binary} (base two) begins with
-\tcode{0b} or \tcode{0B} and consists of a sequence of binary digits.
-An \defnx{octal integer literal}{literal!octal}
-(base eight) begins with the digit \tcode{0} and consists of a
-sequence of octal digits.\footnote{The digits \tcode{8} and \tcode{9} are not octal digits. }
-A \defnx{decimal integer literal}{literal!decimal}
-(base ten) begins with a digit other than \tcode{0} and
-consists of a sequence of decimal digits.
-A \defnx{hexadecimal integer literal}{literal!hexadecimal}
-(base sixteen) begins with
-\tcode{0x} or \tcode{0X} and consists of a sequence of hexadecimal
-digits, which include the decimal digits and the letters \tcode{a}
-through \tcode{f} and \tcode{A} through \tcode{F} with decimal values
-ten through fifteen.
+In an \grammarterm{integer-literal},
+the sequence of
+\grammarterm{binary-digit}s,
+\grammarterm{octal-digit}s,
+\grammarterm{digit}s, or
+\grammarterm{hexadecimal-digit}s
+is interpreted as a base-$N$ number as shown in table \tref{lex.icon.base};
```

Comment:
Remove the hyphen here to avoid confusion with base -N, which is very different from what we mean :) Also maybe "integer" rather than "number"?

Reply:
It's a compound adjective. Plus the "minus" is not in math font, so we're good. (Is there a difference to start with?) Maybe "as an integer of base $N$" to avoid the adjective? Or "as an integer represented with base $N$"?